### DISCUSSION TITLE
Why the error messages are so big/long?

### DISCUSSION THREAD

**Discussion by iuryt** [CliMA/Oceananigans.jl#2408] [General]
Frequently I get huge error logs for simple problems. Is this supposed to be like that? If so, why?

**Comment by glwagner** [comment]
There are two reasons:

1. KernelAbstractions / Cassette (this might get better when we upgrade to KernelAbstractions 0.8)
2. Our types are massive and complex, and Julia's error messages are super precise so list every.single.piece of information.

**Reply by glwagner** [reply]
There's another discussion related to this: #2366

**Reply by iuryt** [reply]
I couldn't tag this as a general question, but I think that your comment is an answer.
